packages/web-api-scan-runner/src/runner/runner.ts

```diff
@@ -118,6 +121,64 @@ export class Runner {
         this.logger.logInfo('Page scan task completed.');
     }
 
+    private async updateWebsiteScanResultWithDiscoveredUrls(
+        pageScanResult: OnDemandPageScanResult,
+        newlyDiscoveredUrls: string[],
+    ): Promise<void> {
+        await this.retryHelper.executeWithRetries(
+            async () => this.updateWebsiteScanResultWithDiscoveredUrlsImpl(pageScanResult, newlyDiscoveredUrls),
+            async (error: Error) => {
+                this.logger.logError(`Failure to generate combined scan result. Retrying on error.`, {
+                    error: System.serializeError(error),
+                });
+            },
+            this.maxCombinedResultProcessingRetryCount,
+            1000,
+        );
+    }
+
+    private async updateWebsiteScanResultWithDiscoveredUrlsImpl(
+        pageScanResult: OnDemandPageScanResult,
+        newlyDiscoveredUrls: string[],
+    ): Promise<void> {
+        if (pageScanResult.websiteScanRefs === undefined || pageScanResult.websiteScanRefs.length === 0) {
+            return;
+        }
+
+        const websiteScanRef = pageScanResult.websiteScanRefs.find((ref) => ref.scanGroupType === 'consolidated-scan-report');
```

Comment:
for crawls, `scanGroupType` should be `deep-scan`

```suggestion
        const websiteScanRef = pageScanResult.websiteScanRefs.find((ref) => ref.scanGroupType === 'deep-scan');
```